internal/provider/destination_resource.go

```diff
@@ -159,7 +159,13 @@ func (r *DestinationResource) Update(ctx context.Context, req resource.UpdateReq
 		return
 	}
 
-	destination, err := r.client.Destinations().Update(ctx, models.DestinationResourceToAPIModel(data))
+	destination := models.DestinationResourceToAPIModel(data)
+	if err := r.client.Destinations().TestConnection(ctx, destination); err != nil {
+		resp.Diagnostics.AddError("Unable to connect to Destination", err.Error())
```

Comment:
Is the `Unable to connect to destination` + `failed to connect to destination` combo going to be a little redundant?

Reply:
Oh yeah, this should probably be `Unable to Update Destination` for consistency (and to make it clear that we didn't update it)

Reply:
Yep, better:
![Screenshot 2024-08-20 at 2 43 57 PM](https://github.com/user-attachments/assets/f3f0fea9-4e5d-4ddf-8ed7-fa08de2804d9)